source/code/include/playfab/PlayFabEventPipeline.h

```diff
@@ -87,6 +87,7 @@ namespace PlayFab
         // We are using that feature (custom pointer relay) because we need to know which batch it was when we receive a callback from the Events API.
         // To keep track of all batches currently in flight (i.e. those for which we called Events API) we need to have a container with controllable size
         // that would allow to quickly map a pointer (like void* customData) to a batch (like a std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>).
+        std::mutex inFlightMutex;
         std::unordered_map<void*, std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>> batchesInFlight;
         std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>> batch;
```

Comment:
I recommend removing `batch` as a member variable and just turning it into a stack variable local to the worker thread.
The easiest way to reduce multi-threading bugs around shared state is to eliminate the possibility of shared state :)

by making this variable local to the worker thread and just passing it into SendBatch (that's the only function that I see using it), you can guarantee that access to it is threadsafe.

```
    void PlayFabEventPipeline::WorkerThread()
    {
        std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>> batch;
        batch.reserve(...);
        // rest of worker thread
```

```
   // have SendBatch accept the worker thread stack batch variable directly instead of referencing it via "this" pointer
    void PlayFabEventPipeline::SendBatch(std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>& batchsize_t& batchCounter) { ... }
```
```

Reply:
resolved!